File: backend/src/main/java/com/ocp/at/entity/Notification.java
package com.ocp.at.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@ToString
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @EqualsAndHashCode.Include
    private String id;

    @Column(nullable = false)
    private String titre;

    @Column(nullable = false)
    private String message;

    @Column(nullable = false)
    @Builder.Default
    private LocalDateTime dateCreation = LocalDateTime.now();

    private LocalDateTime dateLecture;

    @Builder.Default
    private boolean lu = false;

    private String type;

    private String lien;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "utilisateur_id")
    @ToString.Exclude
    private Utilisateur utilisateur;
}

File: mobile/lib/features/notifications/data/notification.dart
/// Modèle Notification — reflète NotificationResponse du backend.
library;

class Notification {
  final String id;
  final String? titre;
  final String? message;
  final DateTime? dateCreation;
  final DateTime? dateLecture;
  final bool lu;
  final String? type;
  final String? lien;
  final String? utilisateurId;

  const Notification({
    required this.id,
    this.titre,
    this.message,
    this.dateCreation,
    this.dateLecture,
    required this.lu,
    this.type,
    this.lien,
    this.utilisateurId,
  });

  factory Notification.fromJson(Map<String, dynamic> json) => Notification(
        id: json['id'] as String,
        titre: json['titre'] as String?,
        message: json['message'] as String?,
        dateCreation:
            json['dateCreation'] == null ? null : DateTime.tryParse(json['dateCreation'].toString()),
        dateLecture:
            json['dateLecture'] == null ? null : DateTime.tryParse(json['dateLecture'].toString()),
        lu: json['lu'] as bool? ?? false,
        type: json['type'] as String?,
        lien: json['lien'] as String?,
        utilisateurId: json['utilisateurId'] as String?,
      );
}
